tools/lsp/preview.rs

```diff
@@ -167,6 +167,45 @@ pub fn config_changed(config: PreviewConfig) {
     };
 }
 
+pub fn adjust_selection(url: VersionedUrl, start_offset: u32, end_offset: u32, new_length: u32) {
+    let Some((version, _)) = get_url_from_cache(url.url()) else {
+        return;
+    };
+    if &version != url.version() {
```

Comment:
But then you need to remove the run_in_ui_thread.
Because still you are testing this in a previous iteration of the event loop, the next iteration might have already changed